### PR TITLE
Push down null filters into TableScan

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/dask-contrib/dask-sql/issues/594

Changes in this PR:

- Implement `DaskTableSource::supports_filter_pushdown` so that it accepts simple `IS NOT NULL` filters
- Change default `nullable` to true

### CODE REVIEW

**Comment by jdye64** [comment]
Ye olde pre-commit got you =)

File: dask_planner/src/sql/table.rs [hunk 1]
```diff
@@ -36,6 +37,36 @@ impl TableSource for DaskTableSource {
...
+    fn supports_filter_pushdown(
+        &self,
+        filter: &Expr,
+    ) -> datafusion_common::Result<TableProviderFilterPushDown> {
+        let mut filters = vec![];
+        split_conjunction(filter, &mut filters);
+        if filters.iter().all(|f| is_supported_push_down_expr(*f)) {
+            // TODO this should return Exact but we cannot make that change until we
+            // are actually pushing the TableScan filters down to the reader because
+            // returning Exact here would remove the Filter from the plan
```
Comment (ayushdg) [inline dask_planner/src/sql/table.rs:52]: Is this TODO an action item on the dask-sql side?

File: dask_planner/src/sql.rs [hunk 2]
```diff
@@ -73,7 +73,7 @@ impl ContextProvider for DaskSQLContext {
                         let mut fields: Vec<Field> = Vec::new();
                         // Iterate through the DaskTable instance and create a Schema instance
                         for (column_name, column_type) in &table.columns {
-                            fields.push(Field::new(column_name, column_type.data_type(), false));
+                            fields.push(Field::new(column_name, column_type.data_type(), true));
```
Comment (jdye64) [inline dask_planner/src/sql.rs:76]: Makes a lot of sense given what Spark SQL does

**Review by ayushdg — APPROVED** [review]
Small question for my understanding but changes look good

File: dask_planner/src/sql/table.rs (hunk 1 above)
Comment (andygrove) [reply dask_planner/src/sql/table.rs:52]: Yes. With this change, the logical plan will now have filters pushed down to the `TableScan` and those need mapping over to the actual read operation.

File: dask_planner/src/sql/table.rs [hunk 3]
```diff
@@ -36,6 +37,36 @@ impl TableSource for DaskTableSource {
...
+            Ok(TableProviderFilterPushDown::Inexact)
+        } else {
+            Ok(TableProviderFilterPushDown::Unsupported)
+        }
+    }
+}
+
+fn is_supported_push_down_expr(expr: &Expr) -> bool {
+    match expr {
+        // for now, we just attempt to push down simple IS NOT NULL filters on columns
```
Comment (jdye64) [inline dask_planner/src/sql/table.rs:66]: Yes, while we can handle more than this I think it makes a lot of sense to handle this low hanging fruit situation. I added issue https://github.com/dask-contrib/dask-sql/issues/596 to cover the remaining situations. That can be handled in a follow up PR

**Review by jdye64 — APPROVED** [review]
LGTM

File: dask_planner/src/sql/table.rs (hunk 1 above)
Comment (jdye64) [reply dask_planner/src/sql/table.rs:52]: https://github.com/dask-contrib/dask-sql/issues/596 made an issue for this FYI

**Pull request merged**